### ISSUE TITLE
Save noise profile as plugin state

### ISSUE THREAD

**Issue by FrnchFrgg** [lucianodato/noise-repellent#9]
Currently, if you make the plugin learn the noise profile all works fine. But if you save/reload the session, then the noise profile is lost and has to be learned anew. This can be tedious if several tracks have noise.

A workaround is to bounce the result of noise repellent, but that essentially freezes the settings, and it's then tedious to revert to the noisy recording (having made sure it is kept somewhere), relearn the noise, reapply and freeze.

A solution is to save the noise profile as plugin state when asked to by the host through LV2.

**Comment by lucianodato** [comment]
Hello Julien! I'm very aware of what you are talking and I will implement this as soon as I can but I've been very busy lately. I assume this will be a matter of few lines of code since I've read the documentation. Cheers!

**Comment by FrnchFrgg** [comment]
No problem. I was just making that into a proper issue so that it can be kept track of. There are several plugins that need to save more state than the control ports can give, and use an LV2 mean to store/load arbitrary data as state (a good example is probably setBfree, also LinuxSampler does that but it's probably less well written)

**Comment by unfa** [comment]
What I did is set up a "learning phase" on my tracks that use Noise Repellent. The first few seconds of the track contains only noise.

I automated Noise Repellent to learn on this part, and stop before the noise ends. Also I automate track's "mute" property to prevent the noise be audible, but it this phase is before the actual session starts - it wouldn't be a problem. I just didn't have room, so I used empty space on the tracks, while other tracks are already playing sounds.

Good think is, if you first do the automation for Noise Repellent on one track, if you copy the plugin to another - the automation will be copied too! This workaround can be pretty fast to apply once you've go the hang of it.
